lib/config/configuration.go

```diff
@@ -1405,6 +1410,63 @@ func applyWindowsDesktopConfig(fc *FileConfig, cfg *service.Config) error {
 	return nil
 }
 
+// applyTracingConfig applies file configuration for the "tracing_service" section.
+func applyTracingConfig(fc *FileConfig, cfg *service.Config) error {
+	// Tracing is enabled.
+	cfg.Tracing.Enabled = true
+
+	if fc.Tracing.ExporterURL == "" {
+		return trace.BadParameter("tracing_service is enabled but no exporter_url is specified")
+	}
+
+	cfg.Tracing.ExporterURL = fc.Tracing.ExporterURL
+	cfg.Tracing.SamplingRatePerMillion = float64(fc.Tracing.SamplingRatePerMillion) / float64(1000000)
+
+	for _, p := range fc.Tracing.KeyPairs {
+		// Check that the certificate exists on disk. This exists to provide the
+		// user a sensible error message.
+		if !utils.FileExists(p.PrivateKey) {
+			return trace.NotFound("tracing_service private key does not exist: %s", p.PrivateKey)
+		}
+		if !utils.FileExists(p.Certificate) {
+			return trace.NotFound("tracing_service cert does not exist: %s", p.Certificate)
+		}
+
+		certificateChainBytes, err := utils.ReadPath(p.Certificate)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		certificateChain, err := utils.ReadCertificateChain(certificateChainBytes)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		if !utils.IsSelfSigned(certificateChain) {
+			if err := utils.VerifyCertificateChain(certificateChain); err != nil {
```

Comment:
I don't think this is the case - that function verifies the cert chain (which SHOULD NOT include the root, because that'd be meaningless) against the system verifier, which will result in an error if one attempts to use a client cert signed by a CA that's not in the system pool.